apps/server/src/utils/fileManagement.ts

```diff
@@ -114,3 +109,20 @@ export async function dockerSafeRename(oldPath: PathLike, newPath: PathLike) {
   await copyFile(oldPath, newPath);
   await unlink(oldPath);
 }
+
+/**
+ * finds potential file index number in our (*) format and increments
+ */
+export function incrementProjectNumber(path: string): string {
+  const { dir, name, ext } = parse(path);
+
+  if (!name.endsWith(')')) return join(dir, `${name} (1)${ext}`);
+
+  const openingParenIndex = name.lastIndexOf(' (');
+  if (openingParenIndex === -1) return join(dir, `${name} (1)${ext}`);
+
+  const maybeNumber = Number(name.slice(openingParenIndex + 2, -1));
```

Comment:
question: what if we looked for the index of closing parenthesis and tried to parse the things in between?